### PR TITLE
hdfs.py refactoring: Move out configuration

### PR DESCRIPTION
**Pull request by Tarrasch**
By putting it in a seperate file, it's easier to refactor the rest of
contents of that file.

### CODE REVIEW

**Comment by erikbern** [comment]
nice. iirc there's some legacy stuff in the configuration that we should drop soon

**Comment by Tarrasch** [comment]
Not sure about that. I did some in https://github.com/spotify/luigi/pull/889 I know. We should however use the new configuration system for the `[hadoop]` section.

By the way, I *really* like the new unified system for doing configuration. It's really nice! :)

File: luigi/contrib/hdfs/config.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import os
+import getpass
+
+try:
+    from urlparse import urlparse, urlunparse
+except ImportError:
+    from urllib.parse import urlparse, urlunparse
+
+
+class hdfs(luigi.Config):
```
Comment (ulzha) [inline luigi/contrib/hdfs/config.py]: This leading to the repetitious "hdfs.config.hdfs()" is unfortunate. And how come it has to have a lovercase name?

Comment (Tarrasch) [inline luigi/contrib/hdfs/config.py]: It's the new configuration system of luigi. This is why you get command line parameters for free, like `--hdfs-client` is `--<classname>-<paramname>`

**Pull request merged**